demo/lightning/image-segmentation/dataset.py

```diff
@@ -31,99 +30,68 @@ def __init__(
-        config=dataflux_mapstyle_dataset.Config(),
         storage_client=None,
         **kwargs,
     ):
 
         self.train_transforms = kwargs["transforms"]
         patch_size, oversampling = kwargs["patch_size"], kwargs["oversampling"]
-        images_prefix, labels_prefix = kwargs["images_prefix"], kwargs[
-            "labels_prefix"]
         self.patch_size = patch_size
         self.rand_crop = RandBalancedCrop(patch_size=patch_size,
                                           oversampling=oversampling)
 
         # Dataflux-specific setup.
         self.project_name = project_name
         self.bucket_name = bucket_name
-        self.config = config
-        self.dataflux_download_optimization_params = (
-            dataflux_core.download.DataFluxDownloadOptimizationParams(
-                max_composite_object_size=self.config.max_composite_object_size
-            ))
-        if not storage_client:
-            self.storage_client = storage.Client(
-                project=project_name,
-            )
-
-        # Data listing.
-        self.images = dataflux_core.fast_list.ListingController(
-            max_parallelism=self.config.num_processes,
-            project=self.project_name,
-            bucket=self.bucket_name,
-            sort_results=self.config.
-            sort_listing_results,  # This needs to be True to map images with labels.
-            prefix=images_prefix,
-        ).run()
-        self.labels = dataflux_core.fast_list.ListingController(
-            max_parallelism=self.config.num_processes,
-            project=self.project_name,
-            bucket=self.bucket_name,
-            sort_results=self.config.
-            sort_listing_results,  # This needs to be True to map images with labels.
-            prefix=labels_prefix,
-        ).run()
+
+        self.images_in_bytes = dataflux_mapstyle_dataset.DataFluxMapStyleDataset(
+            project_name=self.project_name,
+            bucket_name=self.bucket_name,
+            config=dataflux_mapstyle_dataset.Config(
+                sort_listing_results=True,
+                prefix=kwargs["images_prefix"],
+            ),
+        )
+
+        self.labels_in_bytes = dataflux_mapstyle_dataset.DataFluxMapStyleDataset(
+            project_name=self.project_name,
+            bucket_name=self.bucket_name,
+            config=dataflux_mapstyle_dataset.Config(
+                sort_listing_results=True,
+                prefix=kwargs["labels_prefix"],
+            ),
+        )
 
     def __len__(self):
-        return len(self.images)
+        return len(self.images_in_bytes)
 
     def __getitem__(self, idx):
         image = np.load(
             io.BytesIO(
-                dataflux_core.download.download_single(
-                    storage_client=self.storage_client,
-                    bucket_name=self.bucket_name,
-                    object_name=self.images[idx][0],
-                )), )
+                self.images_in_bytes.objects[idx],
```

Comment:
please fix: This will break the __getitem__ logic as the "objects" variable in the dataset just refers to the listing results. Instead, you will need `self.images_in_bytes[idx]` where `images_in_bytes` is the dataset.

Reply:
Thanks for catching! Sent out PR #77 to address all comments on this PR. Thanks for reviewing on you day off.

Reply:
Thanks for the quick fix. I scheduled a /sendLater for Monday morning but you beat me to it ;)

Signed off early on Friday so making the time up for review.